src/graph/graphcontext.c

```diff
@@ -276,9 +287,18 @@ const char *GraphContext_GetAttributeString(const GraphContext *gc, Attribute_ID
 }
 
 Attribute_ID GraphContext_GetAttributeID(const GraphContext *gc, const char *attribute) {
-	Attribute_ID *id = raxFind(gc->attributes, (unsigned char *)attribute, strlen(attribute));
+	// If we are already in a critical region, no further locking should be done in this function.
+	bool in_critical_region = QueryCtx_IsLocked();
```

Comment:
I hope this function isn't called multiple times for a given `attribute` during the lifetime of a query